_posts/2016-12-06-the-state-of-babel.md

```diff
@@ -0,0 +1,460 @@
+> Although companies like Facebook take advantage of these kinds of features, they have created codemods to allow easy change.
+
+---
+
+There isn't enough time or resources to maintain each plugin, especially when there are spec updates.
+
+- Some transforms are simply out of date such as decorators. Logan had to port the previous spec [babel-plugin-transform-decorators-legacy](https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy) for Babel 6 and we haven't had anyone able to rewrite it for the updated spec.
+- [babel/babel#3473 - Async iteration proposal](https://github.com/babel/babel/pull/3473) wasn't merged for so long because we just didn't have time to review it. By the time it was merged it had already moved 
+from stage 2 to stage 3.
+
+---
+
+Next we're going to be working with [Daniel Ehrenberg](https://github.com/littledan) on [Private Fields (Stage 2)](https://github.com/tc39/proposal-private-fields) and [Yehuda Katz](https://github.com/wycats) on [Class and Property Decorators (Stage 2)](http://tc39.github.io/proposal-decorators/).
+
+Relevant Issues:
+
+- Maybe we should think about creating a codemod for Stage x proposals at the same time as creating the actual transform.
+- [Private Fields](https://github.com/babel/babel/issues/4408)
+- [Decorators Proposal](https://github.com/babel/babel/issues/2645)
+
+### Maintaining other ecosystem plugins: JSX/Flow
+
+Babel is vital to the React and Flow ecosystems, and we work closely with the relevant teams at Facebook.
+
+- [React plugins](https://babeljs.io/docs/plugins/#react), [React preset](https://babeljs.io/docs/plugins/preset-react/)
+  - These cover the main `JSX` transform, development plugins, and optimizations.
+- [Flow plugin](https://babeljs.io/docs/plugins/transform-flow-strip-types)
+
+Relevant Issue Labels:
+
+- [babel/babel: React issues](https://github.com/babel/babel/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aopen%20label%3A%22react%22%20)
+- [facebook/react: Optimizing Compiler](https://github.com/facebook/react/labels/Component%3A%20Optimizing%20Compiler)
+
+### [babel-preset-env](https://github.com/babel/babel-preset-env) ("autoprefixer" for Babel)
+
+JavaScript compilation is a moving target: There are yearly updates to the spec, browser vendors are constantly updating to that spec, and users may drop support for earlier browsers. At first glance, there doesn't seem to be a fixed target for what we should compile our JavaScript down to.
+
+This is where `babel-preset-env` comes in: it's a Babel preset that automatically determines the correct Babel plugins to use based on the provided environments.
+
+Its goal is both simplicity in use and efficiency in output: you only need to worry about your target environments to be able to take advantage of native code. The preset will decide for you when you need to add or remove a plugin.
```

Comment:
This is extremely minor detail, but: `s/will decide/decides/` (see http://blog.stoyanstefanov.com/technical-writing-checklist/#will, one of my favorite technical writing tips)